### PR TITLE
bug: update formatting of the response for Get Platform User balance

### PR DESCRIPTION
**Pull request by nsstarr**
## What's done
- Formatted the way we pass response to `printTable()` in `getPlatformUserBalance().`
- Added ResponseType and TokenSymbolType, as well as a type guard to the formatted response
- Created ResponseType and TokenSymbolType
- Added type guard to check the type of response.token and conditionally access the 'symbol' property. 
Now when the table is printed, the token column contains the name of the token instead of `Object object`

## How to test

Run `yarn start` in examples/example-node-api, choose **Get Platform User Balance** option in the CLI and pass the credentials.

## Demo (screenshots, recordings, etc.)

Screenshot 2023-06-28 at 15 02 50

### CODE REVIEW

**Review by teimurjan — CHANGES_REQUESTED** [review]
You have to modify the API function, not the example to change the response value (see `packages/api/src/user/user.ts`).

File: packages/api/src/user/types.ts [hunk 1]
```diff
@@ -7,6 +7,10 @@ export interface HasBalanceArgs {
   amount: string
 }
 
+export interface TokenSymbolType {
```
Comment (teimurjan) [inline packages/api/src/user/types.ts:10]: As you can see, we're following some naming conventions in the API types. You have to do the following:

```suggestion
/*
Example balance response is:

  token: {
    uuid: 'fa209a15-c742-5ecd-9793-745016899ad6',
    name: 'Decimals16',
    symbol: 'QDEC16',
    decimals: 16,
    logo: 'https://roll-external-link-icon.s3.amazonaws.com/roll-test-icon/roll_icon.png',
    contractAddress: '0xd5e7214AcF2447c40AC866e9C5195fc0e419bd0B',
    currentSupply: '1999999.9999999983515342',
    totalSupply: '10000000'
  },
  amount: '0.6',
  updatedAt: '2023-06-29T10:00:56.329565Z'
*/

export interface PlatformUserTokenBalanceResponseData {
  // Given the example response above, describe this type
}
```

File: packages/api/src/user/user.ts [hunk 2]
```diff
@@ -113,13 +114,15 @@ export const getPlatformUserBalance = async (
...
+  const response = await call<Response<{ token: TokenSymbolType }>>({
     url: `/v1/platforms/${userType}/users/${platformUserId}/balances/${tokenId}`,
     method: 'GET',
     authorization: true,
   })
 
-  return response.data
+  const token = response.data.token.symbol
+
+  return { ...response.data, token }
```
Comment (teimurjan) [inline packages/api/src/user/user.ts:125]: We should never modify/re-structure response. Instead, you have to return the original response & show it correctly in the example.

Comment (nsstarr) [reply packages/api/src/user/user.ts:125]: Do you suggest changing the formatting of the response  we pass to the `printTable` function in the example instead?

File: packages/api/src/user/types.ts (hunk 1 above)
Comment (nsstarr) [reply packages/api/src/user/types.ts:10]: Thank you! Will do.

File: packages/api/src/user/user.ts [hunk 3]
```diff
@@ -113,13 +114,15 @@ export const getPlatformUserBalance = async (
   call: Call,
   { userType, platformUserId, tokenId }: PlatformUserTokenBalancesArgs,
 ) => {
-  const response = await call<Response<{ token: string }>>({
+  const response = await call<Response<{ token: TokenSymbolType }>>({
```
Comment (diegonzs-roll) [inline packages/api/src/user/user.ts:117]: Hey @nsstarr the problem as @teimurjan already mentioned is over here... We are returning the wrong response type the correct one should be like the one Tim sent you `{ token, amount, createdAT }`. Then you are going to be able to use the correct typed values on the example and print them nicely using printable.

File: .changeset/olive-toes-remain.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@roll-network/api': minor
+---
+
+Change response formatting in Get Platform User Balance so that it displays the token symbol correctly
```
Comment (teimurjan) [inline .changeset/olive-toes-remain.md:5]: The examples are not part of NPM, so we have to write the changelog regarding the changes in the package itself :)

```suggestion
Updated getPlatformUserBalance response type
```

File: packages/api/src/user/user.ts (hunk 3 above)
Comment (nsstarr) [reply packages/api/src/user/user.ts]: Thanks, @diegonzs-roll . I've updated the response type and changed the formatting in the printTable. Everything now prints out nicely.

File: .changeset/olive-toes-remain.md (hunk 4 above)
Comment (nsstarr) [reply .changeset/olive-toes-remain.md:5]: Thanks for pointing it out @teimurjan, I added the changelog earlier as I was making changes in the packages/api. No need for it now.

**Pull request merged**